src/oidcop/oauth2/introspection.py

```diff
@@ -85,11 +85,11 @@ def process_request(self, request=None, release: Optional[list] = None, **kwargs
         except UnknownToken:
             return {"response_args": _resp}
 
-        _token = _context.session_manager.find_token(
-            _session_info["session_id"], request_token)
+        grant = _session_info["grant"]
+        _token = grant.get_token(request_token)
 
         _info = self._introspect(
-            _token, _session_info["client_id"], _session_info["grant"])
```

Comment:
Do we have test if the _session_info doesn't have a grant key?

anyway ILGTM!

Reply:
hmm, this wouldn't fail, but I think that https://github.com/IdentityPython/oidc-op/pull/31/files#diff-315b01569b3a7bd24a653670e83e37c463e9e86cce66153b84931bccb96ed963R83 could raise an error in that case.

Good catch, I'll look into it and try to write some tests.

Reply:
It should not be possible to have a session without a grant.
As the code is right now when a session is created a grant is added to the session.
This is before user consent or application of service policies so the grant will not really allow anything but it's there.